src/components/app/details/appDetails/SourceInfo.tsx

```diff
@@ -225,7 +260,46 @@ export const SourceInfo = ({
             return <AppDetailsDownloadCard params={paramsId} />
         }
     }
-
+    
+    const RenderCDModal = (): JSX.Element | null => {
+        return (
+            <VisibleModal className="" parentClassName="dc__overflow-hidden">
+                <div className="modal-body--cd-material h-100 contains-diff-view">
+                    {loadingDetails ? (
+                        <>
+                            <div className="trigger-modal__header flex right">
+                                <button type="button" className="dc__transparent" onClick={closeCDModal}>
+                                    <CloseIcon />
+                                </button>
+                            </div>
+                            <div style={{ height: 'calc(100% - 55px)' }}>
+                                <Progressing pageLoader size={32} />
+                            </div>
+                        </>
+                    ) : (
+                        <CDMaterial
+                            materialType={MATERIAL_TYPE.inputMaterialList}
+                            appId={appDetails.appId}
+                            envId={appDetails.environmentId}
+                            pipelineId={appDetails.cdPipelineId} // What will pe pipelineId
+                            stageType={DeploymentNodeType.CD}
+                            envName={appDetails.environmentName}
+                            closeCDModal={closeCDModal}
+                            triggerType={TriggerType.Manual}
+                            history={history}
```

Comment:
Can remove history, match and location props from CDMaterials itself and use the hook itself, it was left as prop since earlier it was a class based component and while migration it might have been missed